lldb/include/lldb/Target/ThreadPlanStack.h

```diff
@@ -202,7 +235,23 @@ class ThreadPlanStackMap {
 
 private:
   Process &m_process;
-  using PlansList = std::unordered_map<lldb::tid_t, ThreadPlanStack>;
+  // We don't want to make copies of these ThreadPlanStacks, there needs to be
+  // just one of these tracking each piece of work.  But we need to move the
+  // work from "attached to a TID" state to "detached" state, which is most
+  // conveniently done by having organizing containers for each of the two 
+  // states.
+  // To make it easy to move these non-copyable entities in and out of the
+  // organizing containers, we make the ThreadPlanStacks into unique_ptr's in a 
+  // storage container - m_plans_up_container.  Storing unique_ptrs means we
+  // can then use the pointer to the ThreadPlanStack in the "organizing"
+  // containers, the TID->Stack map m_plans_list, and the detached plans
+  // vector m_detached_plans.
+  
+  using PlansStore = std::vector<std::unique_ptr<ThreadPlanStack>>;
+  PlansStore m_plans_up_container;
+  std::vector<ThreadPlanStack *> m_detached_plans;
+  
+  using PlansList = std::unordered_map<lldb::tid_t, ThreadPlanStack *>;
```

Comment:
I think I mentioned that in another review already, but if we store `ThreadPlanStack *` instead of `ThreadPlanStack &` shouldn't we check for nullptr at each use? In other words, _could_ we store `ThreadPlanStack &` here?

Reply:
Yeah, I tried making these & not * but it seemed like that involved a lot more fighting with the compiler.  This one is more straightforward.  We will never have nullptr's in this list or return them so I'll add a comment to that effect.

Reply:
I added a comment to the effect that the vector you get of ThreadPlanStack's will never have any nullptr entries.  That is true by the way we deal with them.